dtls/session.go

```diff
@@ -71,6 +71,10 @@ func (s *Session) Context() context.Context {
 	return s.ctx
 }
 
+func (s *Session) SetContextValue(key interface{}, val interface{}) {
```

Comment:
Before s.ctx variable points to same address for whole time, with the new feature we need to guarded with lock.
eg s.ctx as atomic.Value